### ISSUE TITLE
Memory leak on Linux in 1.2.2 (related to URLComponents)

### ISSUE THREAD

**Issue by djones6** [IBM-Swift/Kitura#901]
We are leaking memory on Linux again for Hello World.  It looks like the recent changes which introduced URLComponents (#859 and IBM-Swift/Kitura-net#142) have triggered it.

The same code doesn't seem to leak on Mac, so this could be a Linux Foundation bug.

**Comment by na-gupta** [comment]
Yes, it appears URLComponents and NSURLComponents both currently have leaks. URL however does not appear to leak. We could deprecate urlComponents and make it a lazy var and use URL instead. It's not a good solution, but getting NSURLComponents fixed in Foundation may take some time. @ianpartridge @shmuelk Your thoughts?

**Comment by na-gupta** [comment]
I've tested with the changes above in both linux and mac and everything looks good now.
@djones6 Would appreciate it if you could pull the two PRs and confirm. Thanks!

**Comment by djones6** [comment]
So the two PRs as they stand do resolve the leak for Hello World, however I measured a throughput regression:
```
               | Throughput (req/s)      | CPU (%) | Mem (kb)     | Latency (ms)
Implementation | Average    | Max        | Average | Avg peak RSS | Average  | 99%      | Max
---------------|------------|------------|---------|--------------|----------|----------|----------
      baseline |    44381.5 |    44985.1 |    88.0 |      1109124 |      3.2 |     14.9 |    202.9
         fixes |    38724.2 |    39524.0 |    98.9 |        22000 |      4.2 |     20.0 |    208.7
```
I don't have any insight into the cause of this regression yet.

**Comment by djones6** [comment]
For reference, I raised https://bugs.swift.org/browse/SR-3481 for the memory leak in URLComponents.
